session-state-machine/src/main/scala/zio/raft/sessionstatemachine/SessionStateMachine.scala

```diff
@@ -0,0 +1,530 @@
+          
+          // Add server requests and assign IDs
+          val (stateWithRequests, assignedRequests) = addServerRequests(
+            cmd.createdAt,
+            newState,
+            cmd.sessionId,
+            serverRequests
+          )
+          
+          // Cache the response using composite key
+          val finalState = stateWithRequests.updated["cache"]((cmd.sessionId, cmd.requestId), response)
+          
+          ((response, assignedRequests), finalState)
+    }
+  
+  /**
+   * Handle ServerRequestAck with cumulative acknowledgment.
+   * 
+   * Acknowledging request N removes all pending requests with ID ≤ N.
+   */
+  private def handleServerRequestAck(cmd: SessionCommand.ServerRequestAck[SR]): State[HMap[Schema], Unit] =
+    State.update { state =>
+      acknowledgeServerRequests(state, cmd.sessionId, cmd.requestId)
+    }
+  
+  /**
+   * Handle CreateSession command.
+   */
+  private def handleCreateSession(cmd: SessionCommand.CreateSession[SR]): State[HMap[Schema], List[ServerRequestWithContext[SR]]] =
+    State.modify { state =>
+      // Create session metadata using createdAt from command
+      val metadata = SessionMetadata(
+        capabilities = cmd.capabilities,
+        createdAt = cmd.createdAt
+      )
+      
+      val stateWithMetadata = state.updated["metadata"](
+        cmd.sessionId,
+        metadata
+      )
+      
+      // Initialize lastServerRequestId for this session
+      val stateWithRequestId = stateWithMetadata.updated["lastServerRequestId"](
+        cmd.sessionId,
+        RequestId.zero
+      )
+      
+      // Call user's session created handler (passing capabilities and createdAt)
+      val (serverRequestsLog, newState, _) = handleSessionCreated(cmd.sessionId, cmd.capabilities, cmd.createdAt).runStateLog(stateWithRequestId)
+      val serverRequests = serverRequestsLog.toList  // Convert Chunk[SR] to List[SR]
+      
+      // Add any server requests
+      val (finalState, assignedRequests) = addServerRequests(
+        cmd.createdAt,
+        newState,
+        cmd.sessionId,
+        serverRequests
+      )
+      
+      (assignedRequests, finalState)
+    }
+  
+  /**
+   * Handle SessionExpired command (internal name to avoid conflict with abstract method).
+   */
+  private def handleSessionExpired_internal(cmd: SessionCommand.SessionExpired[SR]): State[HMap[Schema], List[ServerRequestWithContext[SR]]] =
+    State.modify { state =>
+      // Get capabilities from metadata before expiring
+      val capabilities = state.get["metadata"](cmd.sessionId)
+        .map(_.capabilities)
+        .getOrElse(Map.empty[String, String])
+      
+      // Call user's session expired handler with capabilities and createdAt
+      val (serverRequestsLog, stateWithUserChanges, _) = handleSessionExpired(cmd.sessionId, capabilities, cmd.createdAt).runStateLog(state)
+      // Note: serverRequests are ignored - session is expired so we don't send them
```

Comment:
this is a bug, we must add those serverRequestsLog, they are not necessarily for this session.